### PR TITLE
Add sqlite with moor

### PR DESCRIPTION
**Pull request by mysticeti**

### CODE REVIEW

File: lib/database/database_page.dart [hunk 1]
```diff
@@ -162,11 +327,14 @@ class _DatabasePageState extends State<DatabasePage>
...
-    PersonDao personDao = Provider.of<PersonDao>(context);
-    addData(personDao, localizations);
+
+    _personDaoSembast = Provider.of<PersonDaoSembast>(context);
+    addDataSembast(_personDaoSembast, localizations);
+
+    if (!kIsWeb) {
+      _personDaoMoor = Provider.of<PersonDaoMoor>(context);
+      addDataMoor(_personDaoMoor, localizations);
+    }
```
Comment (joykuotw) [inline lib/database/database_page.dart:337]: addDataSembast and addDataMoor are called all the time. As you can see, moor keeps querying which should be avoided.
Screenshot 2021-06-30 at 00 18 50

File: lib/database/database_page.dart [hunk 2]
```diff
@@ -99,9 +144,90 @@ class _DatabasePageState extends State<DatabasePage>
...
+      );
+    }
+  }
+
   @override
   void initState() {
-    _tabController = TabController(vsync: this, length: 1);
+    _tabController = TabController(vsync: this, length: databaseTabs.length);
+    _tabController.addListener(_setActiveTabIndex);
+    _setActiveTabIndex();
```
Comment (joykuotw) [inline lib/database/database_page.dart:230]: why call `_setActiveTabIndex()` again?

**Review by joykuotw — CHANGES_REQUESTED** [review]

File: lib/database/database_page.dart (hunk 2 above)
Comment (mysticeti) [reply lib/database/database_page.dart:230]: To initialize the FAB onPressed function. In the beginning it is an empty function

**Review by joykuotw — APPROVED** [review]

**Comment by joykuotw** [comment]
not necessary to use consumer as there is already provider.of available.
Merge this for now and I will update it in the next commit.

**Pull request merged**